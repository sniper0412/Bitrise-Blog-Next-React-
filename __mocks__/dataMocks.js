export const mockAuthors = [
  {
    slug: 'author-1',
    firstName: 'Bender',
    lastName: 'Rodr√≠guez',
    profileImage: 'http://bender.img',
    title: 'Chief Bending Robot',
    bio: 'A fictional character who is one of the main characters in the animated television series Futurama'
  }
];

export const mockCategories = [
  {
    slug: 'category-1',
    name: 'category 1'
  },
  {
    slug: 'category-2',
    name: 'category 2'
  }
];

export const mockTags = [
  {
    slug: 'tag-1',
    name: 'tag 1'
  },
  {
    slug: 'tag-2',
    name: 'tag 2'
  }
];

export const mockPosts = [
  {
    author: mockAuthors[0],
    slug: 'posts-1',
    seoTitle: 'post 1',
    metaDescription: 'Post 1 description',
    featuredImage: 'http://post-1.img',
    title: 'Post 1',
    body: '<h1>Post 1 body</h1>',
    categories: [mockCategories[0], mockCategories[1]],
    tags: [mockTags[0], mockTags[1]],
    published: 'Wed Jan 02 2019 16:50:20 GMT+0100',
    summary: 'Post 1 summary'
  }
];
