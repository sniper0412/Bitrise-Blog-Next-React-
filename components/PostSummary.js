import React from 'react';
import PropTypes from 'prop-types';

import { PostType } from '../types';
import { formatDate } from '../utils';

const PostSummary = ({ slug, title, summary, categories = [], published, featuredImage, defaultImagePath }) => (
  <div className="article-container" key={slug}>
    <a href={`/post/${slug}`} title={title}>
      <div className="header-container">
        {categories.length > 0 && <p className="category">{categories[0].name}</p>}
        <p className="date">{formatDate(published)}</p>
      </div>

      <div className="image-container" style={{ backgroundImage: `url(${featuredImage || defaultImagePath})` }} />

      <div className="article">
        <h2>{title}</h2>
        <p>{summary}</p>
      </div>
    </a>
  </div>
);

PostSummary.propTypes = { ...PostType, defaultImagePath: PropTypes.string.isRequired };

export default PostSummary;
