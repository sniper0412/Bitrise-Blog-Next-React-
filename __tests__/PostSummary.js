import renderer from 'react-test-renderer';

import { mockPosts } from 'dataMocks';
import PostSummary from '../components/PostSummary';

describe('PostSummary component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<PostSummary {...mockPosts[0]} defaultImagePath="http://default.img" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with the default image', () => {
    const tree = renderer
      .create(<PostSummary {...mockPosts[0]} featuredImage={null} defaultImagePath="http://default.img" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
