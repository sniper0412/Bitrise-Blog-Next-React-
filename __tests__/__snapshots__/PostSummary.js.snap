// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostSummary component renders correctly 1`] = `
<div
  className="article-container"
>
  <a
    href="/posts/posts-1"
    title="Post 1"
  >
    <div
      className="header-container"
    >
      <p
        className="category"
      >
        category 1
      </p>
      <p
        className="date"
      >
        January 2, 2019
      </p>
    </div>
    <div
      className="image-container"
      style={
        Object {
          "backgroundImage": "url(http://post-1.img)",
        }
      }
    />
    <div
      className="article"
    >
      <h2>
        Post 1
      </h2>
      <p>
        Post 1 summary
      </p>
    </div>
  </a>
</div>
`;

exports[`PostSummary component renders correctly with the default image 1`] = `
<div
  className="article-container"
>
  <a
    href="/posts/posts-1"
    title="Post 1"
  >
    <div
      className="header-container"
    >
      <p
        className="category"
      >
        category 1
      </p>
      <p
        className="date"
      >
        January 2, 2019
      </p>
    </div>
    <div
      className="image-container"
      style={
        Object {
          "backgroundImage": "url(http://default.img)",
        }
      }
    />
    <div
      className="article"
    >
      <h2>
        Post 1
      </h2>
      <p>
        Post 1 summary
      </p>
    </div>
  </a>
</div>
`;
